lightning/src/ln/peer_handler.rs

```diff
@@ -1086,9 +1086,8 @@ impl<Descriptor: SocketDescriptor, CM: Deref> PeerManager<Descriptor, CM> where
 					}
 				}
 
-				// check if the peer is ready for encryption
+				// The peer needs to complete its handshake before we can exchange messages
 				if !peer.channel_encryptor.is_ready_for_encryption() {
-					// let's wait for the peer to complete the handshake
 					return true;
 				}
```

Comment:
I'd consider moving this before line 1078. The rationale being that this is a guard clause so what comes before does not need to be checked if this condition is true. That said, the above condition should be mutually exclusive with this one, so reordering should not affect the behavior.